src/Features/Core/Portable/InitializeParameter/AbstractInitializeMemberFromParameterCodeRefactoringProviderMemberCreation.cs

```diff
@@ -103,15 +115,30 @@ protected override Task<ImmutableArray<CodeAction>> GetRefactoringsForAllParamet
                 var field = CreateField(requireAccessibilityModifiers, parameter, rules, parameterNameParts.BaseNameParts);
                 var property = CreateProperty(requireAccessibilityModifiers, parameter, rules, parameterNameParts.BaseNameParts);
 
-                // Offer to generate either a property or a field.  Currently we place the property
-                // suggestion first (to help users with the immutable object+property pattern). But
-                // we could consider swapping this if people prefer creating private fields more.
-                return ImmutableArray.Create<CodeAction>(
-                    new MyCodeAction(string.Format(FeaturesResources.Create_and_initialize_property_0, property.Name),
-                        c => AddSymbolInitializationAsync(document, parameter, functionDeclaration, method, blockStatementOpt, property, c)),
-                    new MyCodeAction(string.Format(FeaturesResources.Create_and_initialize_field_0, field.Name),
-                        c => AddSymbolInitializationAsync(document, parameter, functionDeclaration, method, blockStatementOpt, field, c)));
+                var fieldAction = new MyCodeAction(string.Format(FeaturesResources.Create_and_initialize_field_0, field.Name),
+                    c => AddSymbolInitializationAsync(document, parameter, functionDeclaration, blockStatementOpt, field, c));
+                var propertyAction = new MyCodeAction(string.Format(FeaturesResources.Create_and_initialize_property_0, property.Name),
+                    c => AddSymbolInitializationAsync(document, parameter, functionDeclaration, blockStatementOpt, property, c));
+
+                // Check if the surrounding parameters are assigned to another field in this class.  If so, offer to
```

Comment:
Would this comment be clearer if it specified that the code just takes just the first sibling parameter that is a field or property and uses that to make the determination of which code action to show first? Right now, the comment makes it seem like if _any_ sibling parameter is assigned to a field, then show the field code action first (which I don't believe is what the actual code is doing?).

Reply:
gotcha.  i can def improve this comment. :)